Mathlib/FieldTheory/IntermediateField.lean

```diff
@@ -707,17 +707,19 @@ theorem toSubalgebra_eq_iff : F.toSubalgebra = E.toSubalgebra ↔ F = E := by
   rfl
 #align intermediate_field.to_subalgebra_eq_iff IntermediateField.toSubalgebra_eq_iff
 
-nonrec theorem eq_of_le_of_finrank_le [FiniteDimensional K L] (h_le : F ≤ E)
+theorem eq_of_le_of_finrank_le [hfin : FiniteDimensional K E] (h_le : F ≤ E)
     (h_finrank : finrank K E ≤ finrank K F) : F = E :=
+  haveI : Module.Finite K (Subalgebra.toSubmodule E.toSubalgebra) := hfin
   toSubalgebra_injective <|
-    Subalgebra.toSubmodule.injective <| eq_of_le_of_finrank_le h_le h_finrank
+    Subalgebra.toSubmodule.injective <| FiniteDimensional.eq_of_le_of_finrank_le h_le h_finrank
 #align intermediate_field.eq_of_le_of_finrank_le IntermediateField.eq_of_le_of_finrank_le
 
-theorem eq_of_le_of_finrank_eq [FiniteDimensional K L] (h_le : F ≤ E)
+theorem eq_of_le_of_finrank_eq [FiniteDimensional K E] (h_le : F ≤ E)
     (h_finrank : finrank K F = finrank K E) : F = E :=
   eq_of_le_of_finrank_le h_le h_finrank.ge
 #align intermediate_field.eq_of_le_of_finrank_eq IntermediateField.eq_of_le_of_finrank_eq
 
+-- TODO: generalize `FiniteDimensional K L` to `FiniteDimensional F L` (not so easy)
```

Comment:
```lean
theorem eq_of_le_of_finrank_le'' [FiniteDimensional F L] (h_le : F ≤ E)
    (h_finrank : finrank F L ≤ finrank E L) : F = E := by
  refine le_antisymm h_le ?_
  intro l hl
  let E₁ : IntermediateField F L := Subfield.toIntermediateField E.toSubfield <|
    fun ⟨f, hf⟩ => h_le hf
  let F₁ : IntermediateField F L := Subfield.toIntermediateField F.toSubfield <|
    fun ⟨_, hf⟩ => hf
  replace h_le : F₁ ≤ E₁ := fun f hf => h_le hf
  show l ∈ F₁
  rw [eq_of_le_of_finrank_le' h_le h_finrank]
  exact hl
```
is not very elegant but it works.

Reply:
> What we probably want in general is the function that turns `E` with the condition `F ≤ E` into a `IntermediateField F L`. (It is the line `let E₁ ...` in my code.)

What about `IntermediateField.of_inclusion`? or `of_le`?

Reply:
I think something like `[extendScalars](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/AlgebraTower.html#AlgHom.extendScalars)` is nice, but I am not sure.

Reply:
I added `extendScalars` and it greatly simplifies the proof. 😂